client/tools/mgr-push/uploadLib.py

```diff
@@ -560,7 +558,14 @@ def _processFile(filename, relativeDir=None, source=None, nosig=None):
             lh.append(sstr(a_pkg.header['arch']))
 
         # Build the header hash to be sent
-        info = {'header': Binary(a_pkg.header.unload()),
+        if filename.endswith("deb"):
+            # The 'header' data in the returned dictionary is used with the --headers option for mgr-push
+            # that have to be deprecated as is no longer needed.
```

Comment:
I don't understand this comment. What needs to be deprecated?

Reply:
It seems that the upload-headers-only feature (--headers) is not useful any more as it this was needed by the up2date tool.

Reply:
I still don't understand the comment as it is written in the code, could you please reword it?


If `hdr` is not needed anymore because it's only for the traditional stack, does is still make sense to have it set to different values depending on the package type?

Reply:
I'm not sure if with the `--header` option `mgr-push` runs parts of the traditional stack or not. It does call functionality in the `server.importlib.packageUpload` module inside the XML-RPC server, where the header extraction for an rpm file takes place. 

When the `--newest` option is used, the `unload()` method for the package header is called. This only works for rpm packages so I had to avoid this to happen for a deb packages. 

I believe most of the information in that `info` dictionary is used precisely when the `--header` option is used. With the  `--newest` option, just `info['nvrea']` is later used.

In order to completely remove this header treatment I reckon we should remove the `--header` option.